libcaf_core/src/scheduled_actor.cpp

```diff
@@ -666,6 +633,13 @@ scheduled_actor::categorize(mailbox_element& x) {
     call_handler(down_handler_, this, dm);
     return message_category::internal;
   }
+  if (auto view = make_typed_message_view<action>(content)) {
+    auto ptr = get<0>(view).ptr();
+    CAF_ASSERT(ptr != nullptr);
+    CAF_LOG_DEBUG("run action");
+    ptr->run();
+    return message_category::internal;
+  }
```

Comment:
Yep, you can send an actor any action and it'll run it. Otherwise you can't schedule arbitrary work on an actor, so this is absolutely a feature we need in CAF. Based on this mechanism, I've already implemented the new flow abstraction that leverages actions for back-pressure signaling. Action is an unsafe type, so the this is meant for internal use only.

Reply:
OK. It's the non-internal use that I worry about.